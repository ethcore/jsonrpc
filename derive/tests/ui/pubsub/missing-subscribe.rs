#[macro_use]
extern crate jsonrpc_derive;
extern crate jsonrpc_core;
extern crate jsonrpc_pubsub;

#[rpc(pubsub, name = "hello")]
pub trait Rpc {
	type Metadata;

	/// Hello subscription
	#[rpc(name = "hello_subscribe", alias("hello_sub"))]
	fn subscribe(&self, Self::Metadata, typed::Subscriber<String>, u64);

	/// Unsubscribe from hello subscription.
	#[rpc(unsubscribe, name = "hello_unsubscribe")]
	fn unsubscribe(&self, Option<Self::Metadata>, SubscriptionId) -> Result<bool>;
}

fn main() {}
