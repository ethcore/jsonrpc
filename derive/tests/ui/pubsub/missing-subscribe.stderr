error: custom attribute panicked
 --> $DIR/missing-subscribe.rs:6:1
  |
6 | #[rpc(pubsub, name = "hello")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: [rpc] encountered error: Can't find subscribe method, expected a method annotated with `subscribe` e.g. `#[rpc(subscribe, name = "hello_subscribe")]`

error: aborting due to previous error

